web_infrastructure/jenkins_plugin.py

```diff
@@ -413,6 +413,12 @@ def install(self):
                 self.params['jenkins_home'],
                 self.params['name']))
 
+        hpi_file = '%s/plugins/%s.hpi' % (
+            self.params['jenkins_home'],
+            self.params['name'])
+        if os.path.isfile(hpi_file):
+            os.remove(hpi_file)
```

Comment:
All this should go after the block of commented by `# Send the installation request`.